aegir does not pass forward args to `tsc` command

Today I used [patch-package](https://github.com/ds300/patch-package) to patch `aegir@36.1.3` for the project I'm working on.

I'm working on building out https://github.com/ipfs-shipyard/js-pinning-service-http-client. For this client, I am generating source code from our api spec using `openapi-generator-cli`. That code does not pass when running `aegir build`: 

```bash
generated/fetch/src/apis/PinsApi.ts:18:5 - error TS6133: 'Failure' is declared but its value is never read.

18     Failure,
       ~~~~~~~
generated/fetch/src/apis/PinsApi.ts:19:5 - error TS6133: 'FailureFromJSON' is declared but its value is never read.

19     FailureFromJSON,
       ~~~~~~~~~~~~~~~
generated/fetch/src/apis/PinsApi.ts:20:5 - error TS6133: 'FailureToJSON' is declared but its value is never read.

20     FailureToJSON,
       ~~~~~~~~~~~~~
generated/fetch/src/apis/PinsApi.ts:22:5 - error TS6133: 'PinFromJSON' is declared but its value is never read.

22     PinFromJSON,
       ~~~~~~~~~~~
generated/fetch/src/apis/PinsApi.ts:26:5 - error TS6133: 'PinResultsToJSON' is declared but its value is never read.

26     PinResultsToJSON,
       ~~~~~~~~~~~~~~~~
generated/fetch/src/apis/PinsApi.ts:29:5 - error TS6133: 'PinStatusToJSON' is declared but its value is never read.

29     PinStatusToJSON,
       ~~~~~~~~~~~~~~~
generated/fetch/src/apis/PinsApi.ts:31:5 - error TS6133: 'StatusFromJSON' is declared but its value is never read.

31     StatusFromJSON,
       ~~~~~~~~~~~~~~
generated/fetch/src/apis/PinsApi.ts:32:5 - error TS6133: 'StatusToJSON' is declared but its value is never read.

32     StatusToJSON,
       ~~~~~~~~~~~~
generated/fetch/src/apis/PinsApi.ts:34:5 - error TS6133: 'TextMatchingStrategyFromJSON' is declared but its value is never read.

34     TextMatchingStrategyFromJSON,
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
generated/fetch/src/apis/PinsApi.ts:35:5 - error TS6133: 'TextMatchingStrategyToJSON' is declared but its value is never read.

35     TextMatchingStrategyToJSON,
       ~~~~~~~~~~~~~~~~~~~~~~~~~~
generated/fetch/src/models/Failure.ts:15:1 - error TS6192: All imports in import declaration are unused.

15 import { exists, mapValues } from '../runtime';
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
generated/fetch/src/models/Failure.ts:19:5 - error TS6133: 'FailureErrorFromJSONTyped' is declared but its value is never read.

19     FailureErrorFromJSONTyped,
       ~~~~~~~~~~~~~~~~~~~~~~~~~
generated/fetch/src/models/FailureError.ts:15:18 - error TS6133: 'mapValues' is declared but its value is never read.

15 import { exists, mapValues } from '../runtime';
                    ~~~~~~~~~
generated/fetch/src/models/Pin.ts:15:18 - error TS6133: 'mapValues' is declared but its value is never read.

15 import { exists, mapValues } from '../runtime';
                    ~~~~~~~~~
generated/fetch/src/models/PinResults.ts:15:1 - error TS6192: All imports in import declaration are unused.

15 import { exists, mapValues } from '../runtime';
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
generated/fetch/src/models/PinResults.ts:19:5 - error TS6133: 'PinStatusFromJSONTyped' is declared but its value is never read.

19     PinStatusFromJSONTyped,
       ~~~~~~~~~~~~~~~~~~~~~~
generated/fetch/src/models/PinStatus.ts:15:18 - error TS6133: 'mapValues' is declared but its value is never read.

15 import { exists, mapValues } from '../runtime';
                    ~~~~~~~~~
generated/fetch/src/models/PinStatus.ts:19:5 - error TS6133: 'PinFromJSONTyped' is declared but its value is never read.

19     PinFromJSONTyped,
       ~~~~~~~~~~~~~~~~
generated/fetch/src/models/PinStatus.ts:25:5 - error TS6133: 'StatusFromJSONTyped' is declared but its value is never read.

25     StatusFromJSONTyped,
       ~~~~~~~~~~~~~~~~~~~
generated/fetch/src/apis/PinsApi.ts:18:5 - error TS6133: 'Failure' is declared but its value is never read.

18     Failure,
       ~~~~~~~
generated/fetch/src/apis/PinsApi.ts:19:5 - error TS6133: 'FailureFromJSON' is declared but its value is never read.

19     FailureFromJSON,
       ~~~~~~~~~~~~~~~
generated/fetch/src/apis/PinsApi.ts:20:5 - error TS6133: 'FailureToJSON' is declared but its value is never read.

20     FailureToJSON,
       ~~~~~~~~~~~~~
generated/fetch/src/apis/PinsApi.ts:22:5 - error TS6133: 'PinFromJSON' is declared but its value is never read.

22     PinFromJSON,
       ~~~~~~~~~~~
generated/fetch/src/apis/PinsApi.ts:26:5 - error TS6133: 'PinResultsToJSON' is declared but its value is never read.

26     PinResultsToJSON,
       ~~~~~~~~~~~~~~~~
generated/fetch/src/apis/PinsApi.ts:29:5 - error TS6133: 'PinStatusToJSON' is declared but its value is never read.

29     PinStatusToJSON,
       ~~~~~~~~~~~~~~~
generated/fetch/src/apis/PinsApi.ts:31:5 - error TS6133: 'StatusFromJSON' is declared but its value is never read.

31     StatusFromJSON,
       ~~~~~~~~~~~~~~
generated/fetch/src/apis/PinsApi.ts:32:5 - error TS6133: 'StatusToJSON' is declared but its value is never read.

32     StatusToJSON,
       ~~~~~~~~~~~~
generated/fetch/src/apis/PinsApi.ts:34:5 - error TS6133: 'TextMatchingStrategyFromJSON' is declared but its value is never read.

34     TextMatchingStrategyFromJSON,
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
generated/fetch/src/apis/PinsApi.ts:35:5 - error TS6133: 'TextMatchingStrategyToJSON' is declared but its value is never read.

35     TextMatchingStrategyToJSON,
       ~~~~~~~~~~~~~~~~~~~~~~~~~~
generated/fetch/src/models/Failure.ts:15:1 - error TS6192: All imports in import declaration are unused.

15 import { exists, mapValues } from '../runtime';
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
generated/fetch/src/models/Failure.ts:19:5 - error TS6133: 'FailureErrorFromJSONTyped' is declared but its value is never read.

19     FailureErrorFromJSONTyped,
       ~~~~~~~~~~~~~~~~~~~~~~~~~
generated/fetch/src/models/FailureError.ts:15:18 - error TS6133: 'mapValues' is declared but its value is never read.

15 import { exists, mapValues } from '../runtime';
                    ~~~~~~~~~
generated/fetch/src/models/Pin.ts:15:18 - error TS6133: 'mapValues' is declared but its value is never read.

15 import { exists, mapValues } from '../runtime';
                    ~~~~~~~~~
generated/fetch/src/models/PinResults.ts:15:1 - error TS6192: All imports in import declaration are unused.

15 import { exists, mapValues } from '../runtime';
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
generated/fetch/src/models/PinResults.ts:19:5 - error TS6133: 'PinStatusFromJSONTyped' is declared but its value is never read.

19     PinStatusFromJSONTyped,
       ~~~~~~~~~~~~~~~~~~~~~~
generated/fetch/src/models/PinStatus.ts:15:18 - error TS6133: 'mapValues' is declared but its value is never read.

15 import { exists, mapValues } from '../runtime';
                    ~~~~~~~~~
generated/fetch/src/models/PinStatus.ts:19:5 - error TS6133: 'PinFromJSONTyped' is declared but its value is never read.

19     PinFromJSONTyped,
       ~~~~~~~~~~~~~~~~
generated/fetch/src/models/PinStatus.ts:25:5 - error TS6133: 'StatusFromJSONTyped' is declared but its value is never read.

25     StatusFromJSONTyped,
       ~~~~~~~~~~~~~~~~~~~
```

Because we don't want to maintain some touchy modifications to generated source, I decided to use `tsc-silent` to suppress these trivial errors (unused blah..). However, when I switched the `tsc` command with `tsc-silent` via:

```bash
# Replace aegir's tsc with tsc-silent so we can ignore errors in generated code
mv node_modules/aegir/node_modules/typescript/bin/tsc node_modules/aegir/node_modules/typescript/bin/tsc-bak
ln -fs $(realpath --relative-to=node_modules/aegir/node_modules/typescript/bin $(npm bin)/tsc-silent) node_modules/aegir/node_modules/typescript/bin/tsc
```

And then I ran `aegir build -- -p tsconfig.json --suppress 6133@generated 6192@generated --stats`, I got the following output:

```bash
[12:44:40] Clean ./dist [started]
[12:44:40] Clean ./dist [completed]
[12:44:40] Bundle [started]
[12:44:40] Bundle [completed]
[12:44:40] Generate types [started]
Usage:
  tsc-silent --project <path> [--suppress config | --suppressConfig path] [--compiler path]
             [--watch]

Synopsis:
  --project, -p    Path to tsconfig.json

  --compiler       Path to typescript.js.
                   By default, uses `./node_modules/typescript/lib/typescript.js`.

  --suppress       Suppressed erros.
                   E.g. `--suppress 7017@src/js/ 2322,2339,2344@/src/legacy/`.

  --suppressConfig Path to supressed errors config.
                   See documentation for examples.

  --watch, -w      Run in watch mode.

  --stats          Print number of suppressed errors per path and error code.

Description:
The purpose of the wrapper is to execute TypeScript compiler but suppress some error messages
coming from certain files/folders. For example, this can be used to enable `noImplicitAny` in
some parts of the project while keeping it disabled in others.

[12:44:40] Generate types [failed]
[12:44:40] → Command failed with exit code 1: tsc --build /Users/sgtpooki/code/work/protocol.ai/ipfs/pinning-service-compliance/pinning-service-client/tsconfig-types.aegir.json
Command failed with exit code 1: tsc --build /Users/sgtpooki/code/work/protocol.ai/ipfs/pinning-service-compliance/pinning-service-client/tsconfig-types.aegir.json
```

I noticed that the output passed to `tsc` only included the `--build` option, and none of mine. Investigating further, I was able to fix the problem with the below diff.

## Here is the diff that solved my problem:

```diff
diff --git a/node_modules/aegir/src/build/index.js b/node_modules/aegir/src/build/index.js
index a614f51..e455dfb 100644
--- a/node_modules/aegir/src/build/index.js
+++ b/node_modules/aegir/src/build/index.js
@@ -130,9 +130,7 @@ const tasks = new Listr([
      */
     task: async (ctx, task) => {
       await tsCmd({
-        debug: ctx.debug,
-        tsRepo: ctx.tsRepo,
-        fileConfig: ctx.fileConfig,
+        ...ctx,
         preset: 'types',
         include: ctx.fileConfig.ts.include,
         copyTo: ctx.fileConfig.ts.copyTo,
```

I published a [PR to resolve this issue](https://github.com/ipfs/aegir/pull/939), but I am not sure if the additional properties on `ctx` are unwanted.

<em>This issue body was [partially generated by patch-package](https://github.com/ds300/patch-package/issues/296).</em>

--- a/src/build/index.js
+++ b/src/build/index.js
@@ -130,9 +130,7 @@ const tasks = new Listr([
      */
     task: async (ctx, task) => {
       await tsCmd({
-        debug: ctx.debug,
-        tsRepo: ctx.tsRepo,
-        fileConfig: ctx.fileConfig,
+        ...ctx,
         preset: 'types',
         include: ctx.fileConfig.ts.include,
         copyTo: ctx.fileConfig.ts.copyTo,

